x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorOnly.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.operator;
+
+import org.elasticsearch.transport.TransportRequest;
+
+import java.util.function.Supplier;
+
+public interface OperatorOnly {
+
+    Result check(String action, TransportRequest request);
+
+    enum Status {
+        YES, NO, CONTINUE;
+    }
```

Comment:
I can get rid of the `Status` enum. But if we want to leave spaces for settings (or even partial operator API), the return type needs to be more than just a boolean. Because an error/info message needs to be returned as well for meaningful errors, e.g. exactly which setting(s) is operator-only among all the settings being requested.

Reply:
I think the enum is overkill. An `Optional<String>` would be an OK return type, but it's not perfect.

We use that pattern in `Validatable` in the HLRC (`validate` in transport returns `Exception|null` which is similar)

Reply:
I took a similar approach and it now returns a `Supplier<String>`. I use `Supplier` because the string may not be needed even the user is an operator user. It is only needed for "non-operator-user meets operator-only action".